test/lib/e2e-utils.ts

```diff
@@ -231,6 +231,13 @@ export async function createNext(
 export function nextTestSetup(
   options: Parameters<typeof createNext>[0] & {
     skipDeployment?: boolean
+    /**
+     * If set to `true`, the Next.js instance won't be cleaned up
+     * between multiple tests in the same suite.
+     *
+     * @note Use with caution: Tests should almost never depend on each other's artifacts.
```

Comment:
What would be a legit case? I'm inclined to rename this option to sound more dangerous and make it clear that we want to migrate away.

Reply:
Nothing really, maybe, performance. Some tests run on the same output. Maybe their `expect`s could be consolidated under a single `test` though.